### PR TITLE
added the service endpoints monitoring feature.

### PR DESCRIPTION
**Pull request by Kaali09**

### CODE REVIEW

File: kubernetes/ansible/roles/sunbird-monitoring/defaults/main.yml [hunk 1]
```diff
@@ -215,3 +215,29 @@ redis_process_names: ""
 vault_azure_spark_cluster_http_password: ""
 azure_spark_cluster_http_password: "{{ vault_azure_spark_cluster_http_password }}"
 
+service_health_checks:
+  - service_name: 'analytics'
+    targets:
+      - "http://{{private_ingressgateway_ip}}/analytics/health"
```
Comment (rjshrjndrn) [inline kubernetes/ansible/roles/sunbird-monitoring/defaults/main.yml:221]: this should be the service name
analytics.{{namespace}}.svc

Comment (rjshrjndrn) [reply kubernetes/ansible/roles/sunbird-monitoring/defaults/main.yml:221]: change all the endpoints

File: kubernetes/ansible/roles/sunbird-monitoring/templates/json-path-exporter.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+env: {{ env }}
+
+service_health_checks:
```
Comment (rjshrjndrn) [inline kubernetes/ansible/roles/sunbird-monitoring/templates/json-path-exporter.yaml:3]: remove the dictionary

File: kubernetes/helm_charts/monitoring/json-path-exporter/templates/alertRules.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+---
```
Comment (rjshrjndrn) [inline kubernetes/helm_charts/monitoring/json-path-exporter/templates/alertRules.yaml:1]: Move it to alertrules helm chart

**Review by rjshrjndrn — CHANGES_REQUESTED** [review]

File: kubernetes/ansible/roles/sunbird-monitoring/templates/additional-scrape-configs.yaml [hunk 4]
```diff
@@ -97,4 +97,24 @@ scrapeconfig:
         target_label: instance
       - target_label: __address__
         replacement: blackbox-exporter-prometheus-blackbox-exporter:9115
-  {% endfor %}
+        {% endfor %}
+
+  {% for item in service_health_checks %}
```
Comment (rjshrjndrn) [inline kubernetes/ansible/roles/sunbird-monitoring/templates/additional-scrape-configs.yaml:102]: from where this service _health_checks variable coming ?

File: kubernetes/ansible/roles/sunbird-monitoring/templates/alertrules.yaml [hunk 5]
```diff
@@ -84,3 +84,13 @@ secor_process_names:
 
 redis_process_names:
   {{ redis_process_names | to_yaml | indent( width=2) }}
+
+service_health_checks:
```
Comment (rjshrjndrn) [inline kubernetes/ansible/roles/sunbird-monitoring/templates/alertrules.yaml:88]: define this in roles/defaults/main.yaml and add
service_health_checks: {{ service_health_checks | to_yaml }}

**Review by rjshrjndrn — CHANGES_REQUESTED** [review]

File: kubernetes/ansible/roles/sunbird-monitoring/templates/additional-scrape-configs.yaml (hunk 4 above)
Comment (Kaali09) [reply kubernetes/ansible/roles/sunbird-monitoring/templates/additional-scrape-configs.yaml:102]: @rjshrjndrn From here https://github.com/Kaali09/sunbird-devops/blob/release-3.6.0/kubernetes/ansible/roles/sunbird-monitoring/defaults/main.yml#L218

File: kubernetes/ansible/roles/sunbird-monitoring/templates/alertrules.yaml (hunk 5 above)
Comment (Kaali09) [reply kubernetes/ansible/roles/sunbird-monitoring/templates/alertrules.yaml:88]: Updated as requested.

**Review by rjshrjndrn — APPROVED** [review]

**Pull request merged**